### DISCUSSION TITLE
How to convert GeometryStream to readable class？

### DISCUSSION THREAD

**Discussion by Snailya** [iTwin/itwinjs-core#3552] [Q&A] [answered]
I want to get the brep data of GeometryElement3d element. I think this was stored as properties in GeometryStream.  But the GeometryStream is of Uint8Array, so how to translate into brep data.

The GeometryStream is get by the sql below.
![image](https://user-images.githubusercontent.com/19658624/165233717-2487a17c-ed20-40c2-8404-4ba4924b41df.png)

**Comment by pmconne** [comment, marked as answer]
The GeometryStream property is compressed. You cannot use it in its compressed form as returned by ECSql. You can [iterate a GeometryStream](https://www.itwinjs.org/learning/common/geometrystream/#iterating-a-geometrystream).
That same article also describes (just above "[Creating a GeometryStream](https://www.itwinjs.org/learning/common/geometrystream/#creating-a-geometrystream)") how BRep data is represented. You may be disappointed to learn you cannot do much with BReps using the iTwin.js TypeScript APIs, because useful operations require Parasolid - a proprietary closed-source licensed C library.
What specifically are you trying to do?

**Reply by Snailya** [reply]
I'm now working in a Design Institute which designs assembly line for factory, which means that many steels of different types in my iModel.
One of my work it to count the weight of each type of steels. Currently, this was done by manually put steels on different category.
However, I think that the steel type might be possible classified automatically using machine learning.
To build a classification model, the first is to extract the geometry information as the input data.

p.s. However, I think mesh info is much better now as for some element, there's no brep data. But I'm still want to learn about how to deal with GeometryStream, as that might be used in the future.

**Reply by pmconne** [reply]
If you are trying to extract meshes from elements regardless of how the geometry is actually stored, consider using [exportGraphics](https://www.itwinjs.org/reference/core-backend/imodels/imodeldb/exportgraphics/).
